src/pyconverter/xml2py/ast_tree.py

```diff
@@ -1920,6 +2117,293 @@ class ProductName(Element):
     pass
 
 
+class ArgumentList:
+    def __init__(self, list_entry: VarlistEntry, args: List) -> None:
+
+        self._list_entry = list_entry
+        self._arguments = []
+        self._initial_args = args
+        self._parse_list_entry()
+
+    def _parse_list_entry(self):
+        for item in self._list_entry:
+            if isinstance(item, VarlistEntry):
+                argument_obj = Argument(item, self._initial_args)
+                additional_args = argument_obj.multiple_args
+                if len(additional_args) > 0:
+                    for arg in additional_args:
+                        if arg.py_arg_name != "" and arg.py_arg_name not in self.py_arg_names:
+                            self._arguments.append(arg)
+
+                else:
+                    if argument_obj.py_arg_name != "":
+                        self._arguments.append(argument_obj)
+
+    def __iadd__(self, argument_list):
+        for arg in argument_list.arguments:
+            if arg.py_arg_name not in self.py_arg_names:
+                self._arguments.append(arg)
+        return self
+
+    @property
+    def arguments(self):
+        return self._arguments
+
+    @arguments.setter
+    def arguments(self, argument):
+        self._arguments.append(argument)
+
+    @property
+    def initial_args(self):
+        return self._initial_args
+
+    @property
+    def py_arg_names(self):
+        return [arg.py_arg_name for arg in self._arguments]
+
+
+class Argument:
+    """Argument object."""
+
+    def __init__(
+        self,
+        element: str | Element,
+        initial_argument: List,
+        description: Element | str | None = None,
+    ) -> None:
+        if description is None:
+            if isinstance(element[0], Term):
+                name = element[0]
+                description = element[1]
+
+            else:
+                i = 0
+                while not isinstance(element[i], Term) and i < len(element):
+                    i += 1
+                    name = element[i]
+                    description = element[i + 1]
+
+        else:
+            name = element
+        self._name = name
+        self._description = description
+        self._initial_argument = initial_argument
+
+    @property
+    def is_arg_elipsis(self):
+        """
+        Check if the argument is an elipsis.
+
+        Returns
+        -------
+        bool
+            True if the argument is an elipsis, False otherwise.
+        """
+        return is_elipsis(str(self._name))
+
+    @property
+    def multiple_args(self):
```

Comment:
This function is a bit convoluted... it's hard to follow the logic. As long as it works I'm fine with it given the circumstances